### DISCUSSION TITLE
SQLite?

### DISCUSSION THREAD

**Discussion by ollytheninja** [pretix/pretix#4717] [Q&A] [answered]
Hi there!
I stumbled upon Pretix via the Django Chat podcast, after starting to build my own booking system in Django. As something that I was doing in a volunteer capacity, I'm happy to not have to build the whole thing from scratch - maybe just a couple of plugins.

I noticed the docs say in very big, bold letters NOT to run SQLite in production.
I'm a big fan of Django + SQLite in production, I know a lot has changed in Django since @raphaelm [wrote that warning in 2016](https://github.com/pretix/pretix/commit/f999b9ffd96a441b00ded12cd50ba0cf57ab0593#diff-9787456ededa12b26da5d9215244d1b658c2cc26848a3d6c34e84399b768568dR28) circa Django 1.9.
I'm wondering how relevant that warning is today and whether it's viable to run with SQLite at a small scale (several hundred users)?

**Comment by pc-coholic** [comment, marked as answer]
As one of the developers that is running into the limitations on a regular basis when just doing single person development: Don't.

Just as an example: Payment providers will return a customer after payment and send a webhook at the same moment. Bam, locking issues, 500, sad customer.

**Reply by raphaelm** [reply]
I agree. SQLite in production is a great option nowadays for read-heavy sites. pretix however is a write-heavy site with requirements on transaction isolation and inherent concurrency. I'd avoid putting it to a test.

**Reply by ollytheninja** [reply]
Thank you for the swift response, that makes a lot of sense! I'll plan on running Postgres in production.
Do you know if WAL mode makes it any better?

**Reply by raphaelm** [reply]
The docs say:

> WAL provides more concurrency as readers do not block writers and a writer does not block readers. Reading and writing can proceed concurrently. 

So yes, this will be a lot better than without WAL, as the site can still be rendered while someone buys a ticket. Still, only one write transaction at a time is not a limitation I'd like very much ;)
